### ISSUE TITLE
decorateIcons() accidentally hides polygons from SVG elements

### ISSUE THREAD

**Issue by dicagno** [adobe/helix-project-boilerplate#191]
### Expected Behaviour
SVG images, including vector logos, displaying correctly all the elements (e.g. rects)

### Actual Behaviour
some SVG images are shown with missing rects or polygons

### Root cause (identified)
The following lines are stripping width and height attributes from all the vector elements in a svg element:
https://github.com/adobe/helix-project-boilerplate/blob/main/scripts/lib-franklin.js#LL161-L162

In addition, regexp are used to strip or replace attributes and their values: a safer approach (proposed in the linked PR) would be to use an instance of `DOMParser`.

### Proposed solution
- The PR for this fix removes width and height attributes only from the root <svg> element.
- Usage of DOMParser where possible (as mentioned)

**Comment by ramboz** [comment]
Good point. That regexp is too generic, indeed. The idea was just to strip out the ones on the initial svg tag so that you can essentially adjust the viewport of the SVG via CSS if needed (i.e. to align several icon sizes on a page when the original icons have different sizes)

**Comment by bstopp** [comment]
I think this is no longer an issue given the change icons are handled.

**Comment by fkakatie** [comment]
The current implementation of `decorateIcons()` in [`scripts/aem.js`](https://github.com/adobe/aem-boilerplate/blob/main/scripts/aem.js#L474-L479) no longer uses regex manipulation on SVG markup. The function now creates `<img>` elements that reference SVG icon files, avoiding the issue of stripping width/height attributes from internal SVG elements.

The problematic regex manipulation is no longer present in the codebase. Closing as resolved.
